### PR TITLE
Add musl targets

### LINKED ISSUE #27: Support x86_64-unknown-linux-musl for alpine
I briefly tested using alpine image base today for production docker image. I got this below:

```
#22 22.10 == Compilation error in file lib/wireguardex.ex ==
#22 22.10 ** (RuntimeError) Error while downloading precompiled NIF: precompiled NIF is not available for this target: "x86_64-unknown-linux-musl".        
#22 22.10 The available targets are:
#22 22.10  - aarch64-unknown-linux-gnu
#22 22.10  - x86_64-unknown-linux-gnu
#22 22.10  - aarch64-apple-darwin
#22 22.10  - x86_64-apple-darwin.
#22 22.10
#22 22.10 You can force the project to build from scratch with:
#22 22.10
#22 22.10     config :rustler_precompiled, :force_build, wireguardex: true
#22 22.10
#22 22.10 In order to force the build, you also need to add Rustler as a dependency in your `mix.exs`:
#22 22.10
#22 22.10     {:rustler, ">= 0.0.0", optional: true}
#22 22.10
#22 22.10     lib/wireguardex.ex:13: (module)
#22 22.10     (elixir 1.13.4) lib/kernel/parallel_compiler.ex:346: anonymous fn/5 in Kernel.ParallelCompiler.spawn_workers/7
#22 22.10 could not compile dependency :wireguardex, "mix compile" failed. Errors may have been logged above. You can recompile this dependency with "mix deps.compile wireguardex", update it with "mix deps.update wireguardex" or clean it with "mix deps.clean wireguardex"
```

### PR DESCRIPTION
**Pull request by jamilbk**
Go go gadget `cross-rs`! Was up late and thought I'd see if just adding the targets would work.

Fixes #27

### CODE REVIEW

**Comment by jamilbk** [comment]
Failure being caused by this: https://github.com/rusterlium/rustler/issues/470

**Comment by jamilbk** [comment]
@akrousset (and @conectado) On a slightly related note -- is it convention to use `stable` everywhere in the Rust ecosystem for CI pipelines and projects in general? Or is it more common to lock to a particular version to prevent drift like this?

**Comment by akrousset** [comment]
@jamilbk Yes we should prefer to use stable and only version lock if needed. We could either lock to 1.61 for now or allow the clippy warnings until this fix goes through for Rustler.

**Comment by jamilbk** [comment]
> @jamilbk Yes we should prefer to use stable and only version lock if needed. We could either lock to 1.61 for now or allow the clippy warnings until this fix goes through for Rustler.

That sounds good. I'll try my hand at some Rust today to ignore those clippy linter warnings.

**Comment by conectado** [comment]
> @akrousset (and @conectado) On a slightly related note -- is it convention to use `stable` everywhere in the Rust ecosystem for CI pipelines and projects in general? Or is it more common to lock to a particular version to prevent drift like this?

Seconding what @akrousset said, normally, things like this shouldn't happen and there should be forward compatibility between rust versions. [Editions](https://doc.rust-lang.org/edition-guide/editions/index.html) however would break compatibility.

There's also the concept of [minimum rust version](https://rust-lang.github.io/rfcs/2495-min-rust-version.htm) which I think is not stabilized yet but it's used informally.

File: native/wireguard_nif/src/device.rs [hunk 1]
```diff
@@ -13,7 +13,8 @@ const BACKEND: Backend = Backend::Kernel;
 #[cfg(not(target_os = "linux"))]
 const BACKEND: Backend = Backend::Userspace;
 
-#[allow(clippy::extra_unused_lifetimes)] #[derive(NifStruct)]
+#[allow(clippy::extra_unused_lifetimes)]
```
Comment (akrousset) [inline native/wireguard_nif/src/device.rs]: Oh, right. These are derive macros so the code they output won't be covered by this. 

You could add it globally to the module `#![allow(clippy::extra_unused_lifetimes)]` under the comment at the top.

Comment (jamilbk) [reply native/wireguard_nif/src/device.rs]: Gotcha, was wondering if I was losing my mind.

File: native/wireguard_nif/Cross.toml [hunk 2]
```diff
@@ -1,7 +1,19 @@
...
+[target.aarch64-unknown-linux-musl.env]
+passthrough = [
+  "CARGO_TARGET_AARCH64_UNKNOWN_LINUX_MUSL_RUSTFLAGS"
+]
+
 [build.env]
 passthrough = [
-  "RUSTLER_NIF_VERSION"
+  "RUSTLER_NIF_VERSION",
+  "CARGO_TARGET_X86_64_UNKNOWN_LINUX_MUSL_RUSTFLAGS",
```
Comment (akrousset) [inline native/wireguard_nif/Cross.toml:17]: Probably redundant now?

Comment (jamilbk) [reply native/wireguard_nif/Cross.toml]: Yeah, removed.

**Review by akrousset — APPROVED** [review]
Thanks for the matrix refactor! This LGTM 👍

**Pull request merged**